sdks/python/apache_beam/transforms/trigger.py

```diff
@@ -1368,7 +1368,7 @@ def _output(
     if timestamp is None:
       # If no watermark hold was set, output at end of window.
       timestamp = window.max_timestamp()
-    elif input_watermark < window.end and self.trigger_fn.has_ontime_pane():
+    elif output_watermark < window.end and self.trigger_fn.has_ontime_pane():
```

Comment:
Note that this is the fix in question. Please check @rohdesamuel if that makes sense.

Reply:
Yeah this looks good, thanks for thanks for the fix.